File: 민정/boj18312.py
import sys

fastin = sys.stdin.readline

n, k = map(int, input().split())
cnt = 0
k = str(k)
for i in range(n+1):
    if i<10:
        i = '0' + str(i)
    for j in range(60):
        if j < 10:
            j = '0' + str(j)
        for t in range(60):
            if t < 10:
                t = '0'+str(t)
            if k in str(i) + str(j)+str(t):
                cnt += 1

print(cnt)

File: 세진/Implementation/boj_18312.swift
//
//  main.swift
//  Implementation
//
//  Created by User on 2022/05/15.
//

// https://www.acmicpc.net/problem/18312

import Foundation

let input = readLine()!.split(separator: " ").map({ Int($0)! })
let n = input[0]
let k = input[1]
var cnt = 0

// xx시xx분xx초 형태로 변환 후 k를 포함하는지 확인
func check(hour: Int, minute: Int, second: Int, value: Int) -> Bool {
    var hourString = String(hour)
    var minuteString = String(minute)
    var secondString = String(second)
    
    if hourString.count == 1 { hourString.insert("0", at: hourString.startIndex) }
    if minuteString.count == 1 { minuteString.insert("0", at: minuteString.startIndex) }
    if secondString.count == 1 { secondString.insert("0", at: secondString.startIndex) }
    
    if "\(hourString)\(minuteString)\(secondString)".contains(String(k)) {
        return true
    }
    
    return false
}


for hour in 0...n {
    for minute in 0..<60 {
        for second in 0..<60 {
            // String.contains(value)는 해당 String이 value를 포함하고 있으면 true를 return
            if check(hour: hour, minute: minute, second: second, value: k) {
                cnt += 1
            }
        }
    }
}



print(cnt)
